cpp/src/parquet/variant_test.cc

```diff
@@ -0,0 +1,427 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <gtest/gtest.h>
+
+#include <string>
+
+#include "parquet/exception.h"
+#include "parquet/test_util.h"
+#include "parquet/variant.h"
+
+#include <arrow/filesystem/localfs.h>
+#include <arrow/testing/gtest_util.h>
+#include <arrow/util/base64.h>
+
+namespace parquet::variant {
+
+std::string metadata_test_file_name(std::string_view test_name) {
+  return std::string(test_name) + ".metadata";
+}
+
+std::string value_test_file_name(std::string_view test_name) {
+  return std::string(test_name) + ".value";
+}
+
+std::shared_ptr<::arrow::Buffer> readFromFile(::arrow::fs::FileSystem& fs,
+                                              const std::string& path) {
+  ASSIGN_OR_ABORT(auto file, fs.OpenInputFile(path));
+  ASSIGN_OR_ABORT(auto file_size, file->GetSize());
+  ASSIGN_OR_ABORT(auto buf, file->Read(file_size));
+  return buf;
+}
+
+TEST(ParquetVariant, MetadataBase) {
+  std::string dir_string(parquet::test::get_variant_dir());
+  auto file_system = std::make_shared<::arrow::fs::LocalFileSystem>();
+  std::vector<std::string> primitive_metadatas = {
+      // "primitive_null.metadata",
```

Comment:
Can you please tell me why the null metadata is commented out here?

Reply:
Testing data generated by spark is wrong here, lol: https://github.com/apache/parquet-testing/issues/81